### PR TITLE
Fix some inefficiencies in our build process

### PR DESCRIPTION
**Pull request by kzys**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -340,7 +340,7 @@ $(FIRECRACKER_BIN): $(FIRECRACKER_DIR)/Cargo.toml
 
 .PHONY: firecracker-clean
 firecracker-clean:
-	- $(FIRECRACKER_DIR)/tools/devtool distclean
+	- $(FIRECRACKER_DIR)/tools/devtool clean
```
Comment (swagatbora90) [inline Makefile:343]: Surprisingly I don't see any method named `cmd_clean` in firecracker `main` branch https://github.com/firecracker-microvm/firecracker/blob/main/tools/devtool

It only has a `cmd_distclean` that it supports right now.  Do you have a PR for Firecracker as well?

Comment (kzys) [reply Makefile:343]: You are right.

```
_submodules/firecracker/tools/devtool clean

[31m[Firecracker devtool] Unknown command: clean. Please use `_submodules/firecracker/tools/devtool help` for help.(B[m

make[1]: [Makefile:343: firecracker-clean] Error 1 (ignored)
```

File: Makefile [hunk 2]
```diff
@@ -121,6 +121,7 @@ distclean: clean
 	docker volume rm -f $(GO_CACHE_VOLUME_NAME)
 	$(call rmi-if-exists,$(FIRECRACKER_CONTAINERD_TEST_IMAGE):$(DOCKER_IMAGE_TAG))
 	$(call rmi-if-exists,localhost/$(PROTO_BUILDER_NAME):$(DOCKER_IMAGE_TAG))
+	$(FIRECRACKER_DIR)/tools/devtool distclean
```
Comment (swagatbora90) [inline Makefile:124]: Curious, if we should just call `firecracker-clean` target when someone does `make distclean`?

Comment (kzys) [reply Makefile:124]: `devtool distclean` and `make firecracker-clean`? Sounds good to me.

**Pull request closed without merging**